widget/view.controller.js

```diff
@@ -119,52 +91,134 @@
     }
 
 
+    function _getRegexPattern(indicatorType, regexMapping) {
+      let mapping = {
+        'IP Address': ['IPv4', 'IPv6'],
+        'File Hash': ['MD5', 'SHA1', 'SHA256'],
+        'CIDR Range': ['CIDR'],
+        'Domain': ['Host']
+      };
+      return (mapping[indicatorType] || [indicatorType]).map(key => regexMapping[key]);
+    }
+
+
+    function _getKeyStoreValue(keyStoreName, regexDict) {
+      let [globalVariable, type, defaultKeyStore] = iocExtractionConfigService.constants().globalVariablesToKeyStoreMapping[keyStoreName];
+      return {
+        globalVariable,
+        type,
+        defaultKeyStore,
+        iocValues: [],
+        pattern: _getRegexPattern(type, regexDict)
+      };
+    }
+
+
     function _handleGblVarsAndKeyStores() {
```

Comment:
what do you mean by handle here? please add some comments explaining the function logic

Reply:
Added Comments explaining code, please check